Dockerfile

```diff
@@ -6,19 +6,15 @@
 # of the 'semgrep' wrapping.
 #
 
-FROM ocaml/opam2:alpine@sha256:4c2ce9a181b4b12442a68fc221d0b753959ec80e24eae3bf788eeca4dcb9a293 as build-semgrep-core
+FROM returntocorp/ocaml:alpine as build-semgrep-core
```

Comment:
Can we pin this to a specific hash. We want this public docker image to build as deterministically as possible

Reply:
I was wondering about that. In addition to reproducibility, we want to stay current with the ocaml packages and reduce the amount of maintenance. How about we use the latest image but we print the image ID at runtime (`docker build`) for troubleshooting purposes?

On DockerHub, we could set a tag based on the date e.g. `returntocorp/ocaml:alpine-2020-09-11` in addition to `returntocorp/ocaml:alpine`. I would still reference `returntocorp/ocaml:alpine` in CI builds but a message would indicate that `returntocorp/ocaml:alpine-2020-09-11` is available for troubleshooting purposes.

I'll take care of tagging with dates. This should be helpful to see what changed from one week to another.

Reply:
My worry with having a dynamically updating image behind a single tag is that if someone has returntocorp/ocaml:alpine downloaded on the computer from 5 weeks ago when they docker build again it wont actually pull the latest but use the locally cached one. Not sure what level of caching GithubActions does which might also affect this. Will errors even occur when this happens or is it just a cache miss for dependencies and we end up downloading the needed ones anyway.

Reply:
> if someone has returntocorp/ocaml:alpine downloaded on the computer from 5 weeks ago when they docker build again it wont actually pull the latest

I hadn't realized that. That's a good point.

Reply:
So what I did is the following:

* The generated dockerfile used to build the ocaml base image contains a dummy command with the current date (e.g. `RUN echo 2020-09-11`), which ensure the ocaml image is always fresh.
* Such ocaml images are built weekly.
* Each image is doubly tagged: as just `alpine` and as `alpine-2020-09-11`. See https://hub.docker.com/repository/docker/returntocorp/ocaml

For semgrep's Dockerfile, I suggest using the version like `alpine-2020-09-11`, which won't change and is friendlier than a hash. This makes it easy to communicate with users, and easy for us to see when it gets old and should be updated.

For our CI, I suggest sticking with the latest image such as `alpine`, and it's sufficiently easy to find out which one was used for a given job, by looking at the list of tags on dockerhub.